nephio-webui/set-auth.yaml

```
@@ -0,0 +1,97 @@
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: set-auth
+  annotations:
+    config.kubernetes.io/local-config: "true"
+authProvider: current-context
+oidcTokenProvider: okta
+providerConfig:
+  google:
+    secretName: nephio-google-oauth-client
+  okta:
+    secretName: nephio-okta-oauth-client
+  service-account:
+    secretName: nephio-sa-token
+source: |-
+  load("krmfn.star", "krmfn")
+  def ensure_env(list, env):
+    # looks through the list and adds or updates the env
+    idx = -1
+    for i in range(len(list)):
+      if list[i]["name"] == env["name"]:
+        idx = i
```

I'm not familiar with starlark syntax, but shouldn't be better to break the loop once the item has found
```suggestion
        idx = i
        break
```